### PR TITLE
Milestone 4: Calculator-app logic

### PR DESCRIPTION
**Pull request by enaburekhan**
**Calculator logic in React**

- [ ] In this milestone:

- Create calculate logics using if/else statements
- Create operate logics using if and return statements
- Create constant variables that are wrap in big class in operate class

### CODE REVIEW

File: src/logic/calculate.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
+import operate from './operate';
+
+const isNumber = item => !!item.match(/[0-9]+/);
+
+export default function calculate(data, buttonName) {
```
Comment (bellom) [inline src/logic/calculate.js:3]: According to the instructions on the pathwright, we need to pass an object as an argument.
_Screenshot from [pathwright](https://microverse.pathwright.com/library/react-redux-v11/184438/path/step/106376018/
);_
**image**


The argument passed to the `function calculate` should be an object, we can achieve this by using destructuring.
```
function calculate(data, buttonName) {
  let { total, next, operation } = data;
.....
}
```

File: src/components/App.js [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
 import ButtonPanel from './ButtonPanel';
+import calculate from '../logic/calculate';
 
 function App() {
   return (
     <div className="Component-App">
       <>
         <Display />
         <ButtonPanel />
+        <calculate />
```
Comment (bellom) [inline src/components/App.js:11]: You don't need to add this line, you have done well by adding the `calculate` function on line 3.
kindly remove this line.

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @enaburekhan  :clap: 

### Awesome job so far!! :+1: 

#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 

# Status: Changes Required! ♻️ 

**Kindly check out my comments below;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

File: src/components/App.js (hunk 2 above)
Comment (enaburekhan) [reply src/components/App.js:11]: **The following changes were made:**

- Destructure data parameter into total, next, and operation in calculate class.

- I had to leave line 11 in App.js. When I remove it, I get an undefined error. Besides, Adding it is part of the milestone requirements. see screen-shot for reference.

image

Comment (bellom) [reply src/components/App.js:11]: @enaburekhan 🙂, I get your point.
I think the numbering is confusing, 
let say `Add calculate.js to the App component` is the title
and `import the calculate function is the subtitle.

15 - saying what to do 
16 - saying how to go about it.

kindly remove in the next milestone, i understand the `undefined` error message.

**Review by bellom — APPROVED** [review]
## Hi @enaburekhan  :clap: 

**Great work making all the necessary corrections from the previous review.**

# Status: Approved 👍 

## Congratulation :+1:  
_Go ahead and merge to dev and don't forget to submit the project completion form.
Good luck in the next milestone :slightly_smiling_face:_ 

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Pull request merged**